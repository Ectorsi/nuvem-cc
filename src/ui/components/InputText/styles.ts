import styled from 'styled-components';

export const InputContainer = styled.input`
    border: 1px solid #000;
    border-radius: 4px;
    padding: 8px;
    width: 100%;
    margin-bottom: 16px;
`;

export const InputLabel = styled.label`
    display: block;
    margin-bottom: 8px;

    font-size: 16px;
    font-weight: 600;

    color: #000;
`;

export const Error = styled.span`
    display: block;
    margin-top: 8px;

    font-size: 14px;
    font-weight: 600;

    color: white;
`;
