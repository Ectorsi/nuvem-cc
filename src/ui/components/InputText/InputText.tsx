import React, { ChangeEvent, InputHTMLAttributes } from 'react';
import * as S from './styles';

export type InputTextProps = {
    label: string;
    placeholder: string;
    value: string;
    onChange: (e: ChangeEvent<HTMLInputElement>) => void;
    error?: string;
} & InputHTMLAttributes<HTMLInputElement>;

const InputText = ({ label, placeholder, value, onChange, error, ...rest }: InputTextProps) => {
    return (
        <>
            <S.InputLabel>{label}</S.InputLabel>
            <S.InputContainer
                type="text"
                placeholder={placeholder}
                value={value}
                onChange={onChange}
                {...rest}
            />
            {error && <S.Error>{error}</S.Error>}
        </>
    );
};

export default InputText;