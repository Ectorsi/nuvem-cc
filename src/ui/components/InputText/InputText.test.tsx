import React from "react";
import { render, screen } from "@testing-library/react";

import InputText, { InputTextProps } from './InputText';

const mockInputText: InputTextProps = {
    label: 'Test',
    placeholder: 'Test',
    value: 'Test',
    onChange: jest.fn(),
    error: '',
};

describe('InputText unit test', () => {
    it("should be able to show the InputText's label and input", () => {
        const { container } = render(<InputText {...mockInputText} />);
        const Label = screen.getByText(/Test/i);
        const input = screen.getByPlaceholderText(/Test/i);
        expect(input).toBeInTheDocument();
        expect(Label).toBeInTheDocument();
        expect(container.firstChild).toMatchSnapshot();
    });

    it("should be able to show the InputText's error message", () => {
        render(<InputText {...mockInputText} error="error test" />);
        const errorMessage = screen.getAllByText(/error test/i)[0];
        expect(errorMessage).toBeInTheDocument();
    });
});