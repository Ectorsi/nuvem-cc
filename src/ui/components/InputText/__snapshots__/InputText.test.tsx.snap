// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText unit test should be able to show the InputText's label and input 1`] = `
<label
  class="sc-gueYoa"
>
  Test
</label>
`;
